7task7/src/AdList.js

```diff
@@ -0,0 +1,423 @@
+            if (!item.isDelete) {
+                name.push(item.vendor);
+            }
+        });
+        let unique = AdList.unique(name);
+        return unique;
+    }
+
+
+    get(id) {
+        if (typeof id == "string" && id.valueOf() > 0) {
+            return this._items.find((item) => item.id == id);
+        }
+        return false;
+    }
+
+
+    getPage(skip, top, filterConfig) {
+        let result;
+        if (!filterConfig) {
+            result = this._items;
+        }
+
+        if (filterConfig) {
+
+            if (filterConfig.vendor) {
+                result = this._items.filter(function (item) {
+                        return item.vendor.toLowerCase().indexOf(filterConfig.vendor.toLowerCase()) + 1;
+                    }
+                );
+            }
+
+            if (filterConfig.createdAt) {
+                result = this._items.filter(function (item) {
+                    return item.createdAt >= filterConfig.createdAt;
+                });
+            }
+
+            if (filterConfig.validUntil) {
+                result = this._items.filter(function (item) {
+                    return item.validUntil.getTime() <= filterConfig.validUntil.getTime();
+                });
+            }
+
+
+            if (filterConfig.hashTagSearch && filterConfig.hashTagSearch.length!=0) {
+                result = this._items.filter(function (post) {
+                    if(typeof post.hashTags!== "undefined") {
+                        return filterConfig.hashTagSearch.every(function (tag) {
+                            return post.hashTags.includes(tag);
+                        });
+                    }
+
+                });
+            }
+        }
+        return result.sort((function(a, b) {
+                return a.createdAt-b.createdAt
+        }
+        )).slice(skip, skip+top);
+    }
+
+
+    add(item) {
+        const date = new Date();
+        item.id=+date;
+        item.createdAt = new Date();
+        if (AdList.validateAd(item)) {
+            this._items.push(item);
+            return true;
+        }
+        return false;
+    }
+
+
+    edit(id, adItem) {
+        let num = this._items.findIndex(item => item.id === id);
+
+        if (adItem.description) {
+            this._items[num].description = adItem.description;
+        }
+        if (adItem.link) {
+            this._items[num].link = adItem.link;
+        }
+        if (adItem.discount) {
+            this._items[num].discount = adItem.discount;
+        }
+        if (adItem.validUntil) {
+            this._items[num].validUntil = adItem.validUntil;
+        }
+        if (adItem.hashTags) {
+            this._items[num].hashTags = adItem.hashTags;
+        }
+        if (AdList.validateAd(this._items[num]) === false){
+            return false;
+        }
+        else
+            return true;
+    }
+
+
+    remove(id) {
+        let tmp = this._items.slice();
+        tmp.splice(tmp.findIndex(item => item.id === id), 1);
```

Comment:
If result of `findIndex` is -1 (that means that there's no such element in array), than splice will remove last element. Please add condition before use splice.